client/api.c

```diff
@@ -1624,6 +1624,12 @@ TDNFResolve(
         pSolvedPkgInfo->pPkgsToDowngrade ||
         pSolvedPkgInfo->pPkgsToReinstall;
 
+    dwError = TDNFCalculateTotalDownloadSize(pSolvedPkgInfo);
+    BAIL_ON_TDNF_ERROR(dwError);
+
+    dwError = TDNFCheckAvailableCache(pTdnf->pConf, pSolvedPkgInfo);
```

Comment:
I like this better! I thought maybe having the accumulated size stored somewhere would be useful for something later on. But for now, this is cleaner.

Reply:
For what future requirements we can use this stored total size?

I think my idea is simple and realistic here.

When shopping for something we first think how much amount we have in hand and then later we shop accordingly.
Likewise, first calculate total size available and then check if we crossed it. It might save some iterations when available size is actually lesser than total size of rpms to be downloaded.

It's okay if you want to keep the total download size in a structure member for future needs but design-wise, I'm not really convinced with the existing approach.

Reply:
Oh no I meant what you suggested was much cleaner. I agree with your suggestions and in fact added it to the latest commit earlier today: https://github.com/vmware/tdnf/pull/321/commits/4a7d1aff2cbe1b83473bd13f6867e345cd106d2b. That's why I said I liked your approach better! 

I should have been more clear, sorry about that.

As for a future use case for storing download sizes, I'm not exactly sure. So I removed it as suggested in the same commit mentioned previously: https://github.com/vmware/tdnf/pull/321/commits/4a7d1aff2cbe1b83473bd13f6867e345cd106d2b

Reply:
Understood, thanks for the clarification.